app.rb

```diff
@@ -0,0 +1,142 @@
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'book'
+
+class App
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  def intro
+    puts 'Welcome to School Library App!'
+  end
+
+  def display_options
+    puts 'Please choose an option by entering a number:'
+    puts '1 - List all books'
+    puts '2 - List all people'
+    puts '3 - Create a person'
+    puts '4 - Create a book'
+    puts '5 - Create a rental'
+    puts '6 - List all rentals for a given person id'
+    puts '7 - Exit'
+  end
+
+  # option 1 - List all books
+  def list_books
+    puts 'No book yet!' if @books.empty?
+    @books.each { |book| puts "Title: \"#{book.title}\", Author: #{book.author}" }
+  end
+
+  # option 2 - List all people
+  def list_people
+    puts 'We do not have people yet' if @people.empty?
+    @people.each { |person| puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}" }
+  end
+
+  # Create a student
+  def create_student
+    print 'Age: '
+    age = gets.to_i
+    print "\nName: "
+    name = gets.chomp.capitalize
+    print "\nHas parent permission? [Y/N]: "
+    permission = gets.downcase == 'y'
+    @people << Student.new(age, name, permission)
+    puts 'Person created successfully'
+  end
+
```

Comment:
![image](https://github.com/kkmanuu/OOP-school-library/assets/93039359/5081236d-bc37-4df3-bb1d-167af19843ff)

- [ ] Great work so far but it would be nice if you double-check the behavior of your app :thinking: because I cannot create a student or teacher because of the error which you can see in the image above :arrow_up: Kindly fix this error to make your app work as expected :blush: :+1:

Reply:
@HAMMAS-SALEEM Done making the changes